### PR TITLE
feat(analytics): Add Google PubSub backend

### PR DESCRIPTION
**Pull request by mattrobenolt**
Replaces #5420

Decided that the best way moving forward was to simply write into PubSub instead of directly into BigQuery. PubSub provides being able to natively send batches asynchronously instead of BigQuery streaming inserts being done synchronously.

Having events flow into PubSub also gives us significantly more flexibility like being able to attach multiple subscribers to the data to do different things with it. Like, archiving into Google Storage for long term, and writing in BQ. Or reprocessing old/raw data.

cc @ehfeng

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Review by tkaemming — APPROVED** [review]
I'm approving this since /this/ change itself is fine, but I'm pretty sure some of the existing code it changes is not.

File: src/sentry/analytics/event.py [hunk 1]
```diff
@@ -101,7 +104,8 @@ def __init__(self, type=None, datetime=None, **items):
     def serialize(self):
         return dict(
             {
-                'timestamp': int(self.datetime.isoformat('%s')),
+                'guid': self.guid,
+                'timestamp': int(self.datetime.strftime('%s')),
```
Comment (tkaemming) [inline src/sentry/analytics/event.py]: Shouldn't this use `sentry.utils.dates.to_timestamp`?

I'm not sure if this was ever resolved: https://bugs.python.org/issue12750#msg142249

**Review by tkaemming — COMMENTED** [review]
Forgot to actually include the comment as part of the review. 😑

File: src/sentry/analytics/event.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/analytics/event.py]: TIL

I can switch it to this.

Comment (mattrobenolt) [reply src/sentry/analytics/event.py]: Done in c406b2763cf764c562913fc2e5187063a039ce37

**Review by tkaemming — APPROVED** [review]
Seems faster

**Pull request merged**